axlearn/common/attention.py

```diff
@@ -1847,13 +1939,12 @@ def forward(
 
     def _compute_logits(self, q_proj: Tensor, k_proj: Tensor) -> Tensor:
         cfg = self.config
-        if cfg.per_dim_scale is not None:
-            # Applies a per dim scale on q_proj.
-            q_proj = self.per_dim_scale(q_proj)
+        with child_context("apply_per_dim_scale", module=self):
+            q_proj = self.scale_query.apply_per_dim_scale(q_proj)
```

Comment:
Should we call `scale_query.apply_norm` before `apply_per_dim_scale`?

Reply:
Originally I wasn't going to, but in the end have now added it + a comment to clarify that only the scale_factor is applied to either/or logits/proj.